src/terrain/terrain.js

```diff
@@ -580,34 +598,17 @@ export class Terrain extends Elevation {
         program.setTerrainUniformValues(context, uniforms);
     }
 
-    // If terrain handles layer rendering (rasterize it), return true.
-    renderLayer(layer: StyleLayer, _?: SourceCache): boolean {
```

Comment:
Removed this method and replaced it with batched rendering in the main render loop, the reason is to make the render commands and drawn layer range more explicit while reducing potential branching.

Reply:
nice, this is much clearer 💯